sound/soc/sof/ipc.c

```diff
@@ -268,11 +268,10 @@ int sof_ipc_tx_message(struct snd_sof_ipc *ipc, u32 header,
 
 	spin_unlock_irq(&sdev->ipc_lock);
 
-	/* schedule the message if not busy */
-	if (!sdev->disable_ipc_queue && snd_sof_dsp_is_ipc_ready(sdev))
-		ret = ipc_tx_next_msg(sdev->ipc);
-	else
+	if (sdev->disable_ipc_queue)
 		ret = -EAGAIN;
+	else
```

Comment:
if we disable the ipc_queue, then is -EAGAIN a valid return value? retrying isn't going to be more successful.

Reply:
@lyakh @plbossart there is no queue anymore. So we can just remove disable_ipc_queue

Reply:
@ranj063 @plbossart I agree, that EAGAIN isn't a good choice, I'll replace it with ENODEV. But I actually think that we need that flag. It should be used to wait for a running IPC TX to complete. But for that setting the flag should be guarded by the same mutex. I'll make that change. Thanks for catching!